### PR TITLE
nawk: 20240422 -> 20240728

### PR DESCRIPTION
**Pull request by afh**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by afh** [comment]
Result of `nixpkgs-review pr 332084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nawk</li>
    <li>nawk.man</li>
    <li>noweb</li>
    <li>noweb.tex</li>
  </ul>
</details>

File: pkgs/by-name/na/nawk/package.nix [hunk 1]
```diff
@@ -1,28 +1,19 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, bison
-, buildPackages
-, installShellFiles
-}:
+{ lib, stdenv, fetchFromGitHub, bison, buildPackages, installShellFiles }:
```
Comment (AndersonTorres) [inline pkgs/by-name/na/nawk/package.nix:1]: 1. What is happening with the Nix formatter? Lists like these should be in a form similar to

```nix
{
  lib,
  stdenv,
  fetchFromGitHub,
  . . .
}
```

File: pkgs/by-name/na/nawk/package.nix [hunk 2]
```diff
@@ -1,28 +1,19 @@
...
+    hash = "sha256-LA7fdbMP3aKJ1QljoKWizqVg3ys3hd8tGaRsQnIO+Hc=";
   };
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
 
-  nativeBuildInputs = [
-    bison
-    installShellFiles
-  ];
+  nativeBuildInputs = [ bison installShellFiles ];
```
Comment (AndersonTorres) [inline pkgs/by-name/na/nawk/package.nix:16]: ... and this should be not needed

File: pkgs/by-name/na/nawk/package.nix [hunk 3]
```diff
@@ -47,7 +38,8 @@ stdenv.mkDerivation (finalAttrs: {
       Second Edition, by Al Aho, Brian Kernighan, and Peter Weinberger
       (Addison-Wesley, 2023, ISBN 0-13-826972-6).
     '';
-    changelog = "https://github.com/onetrueawk/awk/blob/${finalAttrs.src.rev}/ChangeLog";
+    changelog =
```
Comment (AndersonTorres) [inline pkgs/by-name/na/nawk/package.nix:41]: ... and it looks like a bug from formatter

File: pkgs/by-name/na/nawk/package.nix (hunk 1 above)
Comment (afh) [reply pkgs/by-name/na/nawk/package.nix:1]: 🤦 I was running `nix run nixpkgs#nixfmt` thinking this was already using the new RFC style…

**Comment by afh** [comment]
`bison` seems to be needed as `nativeBuildInput` as the package fails with `bison: command not found` otherwise. Any idea what else might be going wrong, @AndersonTorres?

File: pkgs/by-name/na/nawk/package.nix (hunk 2 above)
Comment (afh) [reply pkgs/by-name/na/nawk/package.nix:23]: * If `bison` is not in `nativeBuildInputs` the build fails with `bison command not found`
* If `installShellFiles` is not in `nativeBuildInputs` the nixf-tidy check fails due to unused import

Comment (AndersonTorres) [reply pkgs/by-name/na/nawk/package.nix:23]: I was talking about the formatter, not the code.

**Review by AndersonTorres — APPROVED** [review]

**Review by pinage404 — APPROVED** [review]
Result of `nixpkgs-review pr 332084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nawk</li>
    <li>nawk.man</li>
    <li>noweb</li>
    <li>noweb.tex</li>
  </ul>
</details>

I didn't test the package

**Comment by afh** [comment]
Would be be willing to help getting this merged, @JohnRTitor?

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 332084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>nawk</li>
    <li>nawk.man</li>
    <li>noweb</li>
    <li>noweb.tex</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 332084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nawk</li>
    <li>nawk.man</li>
    <li>noweb</li>
    <li>noweb.tex</li>
  </ul>
</details>

**Comment by afh** [comment]
Thanks for much for helping getting this merged, @JohnRTitor, very helpful and very much appreciated 💯

**Pull request merged**